airflow/www_rbac/views.py

```diff
@@ -1123,27 +1144,27 @@ def dagrun_clear(self):
         end_date = execution_date
 
         return self._clear_dag_tis(dag, start_date, end_date, origin,
+                                   event_name='dagrun_clear',
                                    recursive=True, confirmed=confirmed)
 
     @expose('/dagrun_cancel', methods=['POST', 'GET'])
     @has_dag_access(can_dag_edit=True)
     @has_access
-    @action_logging
     def dagrun_cancel(self):
-        # cancels all dagruns...
+        # cancels all dagruns. action_logging is skipped so we can log each dagrun
         dag_id = request.args.get('dag_id')
         origin = request.args.get('origin') or "/"
         confirmed = request.form.get('confirmed') == "true"
 
         dag = dagbag.get_dag(dag_id)
 
         return self._clear_dag_tis(dag, None, None, origin,
+                                   event_name='dagrun_cancel_all',
                                    recursive=True, confirmed=confirmed, cancel=True)
 
     @expose('/cancel_dagrun', methods=['POST'])
     @has_dag_access(can_dag_edit=True)
     @has_access
-    @action_logging
```

Comment:
Why nix this? This is for a specific dagrun? Only one execution date of interest.

Reply:
so idk why but airflow calls each of these cancel/clear functions twice when canceling/clearing. first one is a dry run and second is the actual one. that leads to it being logged twice each time we do the action. by removing it and just having it in the helper, it only gets logged once

Reply:
I see - so the unconfirmed time it will log and then log the same thing again.